orchestrator/lambdas/layers/heimdall_utils/heimdall_utils/utils.py

```diff
@@ -96,18 +100,28 @@ def parse_timestamp(timestamp: Optional[str] = None) -> str:
     Notes:
         - The function uses the current system time to calculate the 3-month offset.
         - All returned timestamps are in UTC (denoted by the 'Z' suffix).
-        - The function logs a warning message when generating a default timestamp.
     """
-    format = "%Y-%m-%dT%H:%M:%SZ"
     try:
-        if timestamp and bool(datetime.strptime(timestamp, format)):
+        if timestamp and is_valid_timestamp(timestamp):
             return timestamp
     except (TypeError, ValueError):
         log.error("Timestamp is invalid")
```

Comment:
Nit: This error message may not be accurate anymore, since (if I understand correctly) a valid timestamp can now fail for being too old.